lib/rules/no-early-return.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @author *****your name*****
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow early returns in setup and data functions',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-early-return.html'
+    },
+    fixable: null,
+    schema: []
+  },
+
+  /** @param {RuleContext} context */
+  create(context) {
+    const rStatementLocs = []
+    return utils.compositingVisitors(
+      utils.defineVueVisitor(context, {
+        ReturnStatement(node) {
+          if (context.getScope().type === 'function') {
+            return
+          }
+          rStatementLocs.push(node.loc)
+        },
+
+        onSetupFunctionExit() {
+          if (rStatementLocs.length === 0) {
+            return
+          }
+
+          for (const loc of rStatementLocs) {
+            context.report({
+              loc,
+              message: 'Extra return statement in setup function.'
+            })
+          }
+        },
+        onVueObjectEnter(node) {
+          const dataProperty = utils.findProperty(node, 'data')
```

Comment:
The same logic should also apply to Nuxt.js's `asyncData` property: https://nuxtjs.org/docs/directory-structure/pages#asyncdata